src/mango/src/mango_idx_nouveau.erl

```diff
@@ -307,7 +307,7 @@ construct_analyzer({Props}) ->
 
 indexable_fields(Selector) ->
     TupleTree = mango_selector_text:convert([], Selector),
-    indexable_fields([], TupleTree).
+    ordsets:to_list(indexable_fields(ordsets:new(), TupleTree)).
```

Comment:
is `ordsets` throughout the function better/faster than just a `lists:ukeysort` at the end?

Reply:
If I recall correctly, this method was chosen based on a discussion on the topic on Slack.  But I might have misunderstood that.  Either way it happened, I have done some simple measurements to see which approach is faster.  Although I believe you meant `lists:usort/1`?  The `lists:ukeysort/2` function seems to work with tuples only.

```console
> Xs = lists:map(fun(_X) -> rand:uniform(1000) end, lists:seq(1, 10_000)).
[153,53,32,94,647,739,807,484,695,434,934,437,149,888,853,
 251,639,101,838,404,236,68,187,350,467,162,1000,775,930|...]
> element(1, timer:tc(fun() -> ordsets:to_list(lists:foldr(fun(X, Acc) -> ordsets:add_element(X, Acc) end, ordsets:new(), Xs)) end)) / 1000.
63.81
> > element(1, timer:tc(fun() -> lists:usort(lists:foldr(fun(X, Acc) -> [X | Acc] end, [], Xs)) end)) / 1000.
11.752
```

The results suggest that the use of `ordsets` is not justified here.

Reply:
Oh, now I remember: the reason for choosing `ordsets` is to maintain the original order of the fields.  A side-effect of `lists:usort/1` is that the elements will be sorted.

Reply:
`lists:uniq/1` could make a good candidate but that requires OTP 25.

Reply:
the docs for ordsets says `Elements are ordered according to the Erlang term order.`. If the intent is to preserve the original order, neither ordsets or usort/ukeysort seems to be right?

Reply:
Oh, this property of Erlang `ordsets` has escaped my attention.  If that is case, you are right.

Reply:
I have pushed a version that uses a version of `lists:uniq/1` to deduplicate the collected elements at the end.